src/field/gf_101_2.rs

```diff
@@ -0,0 +1,335 @@
+use core::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+use std::ops::{Div, DivAssign, MulAssign, Rem};
+
+use serde::{Deserialize, Serialize};
+
+use super::ExtensionField;
+use crate::field::FiniteField;
+
+/// Pluto curve with modulus 101 supports two degree extension field. This can be verified
+/// by finding out embedding degree of the curve, i.e. smallest k: r|q^k-1
+const EXT_DEGREE: usize = 2;
+const QUADRATIC_EXTENSION_FIELD_ORDER: u32 = 101 * 101;
+
+#[derive(Clone, Default, Copy, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
+
+/// Quadratic Extension field element represented as polynomial of degree 1 in form:
+/// a_0 + a_1*t where {a_0, a_1} \in \mathhbb{F}. Uses irreducible poly of the form:
+/// (X^2-K).
```

Comment:
should this be X^2 -K or X^2 + K?  X^2 + 2 is irreducible in F_{101} and should be used to generate our extension field.

Reply:
I think both work as (correct me if I'm wrong) 2 is not a square in GF_{101}.

Reply:
yeah, both of them are irreducible, added it in the field sage script as well

https://github.com/pluto/ronkathon/blob/9c0ba7055b0f80b0d157243ca48bf002ed823ca4/math/field.sage#L47-L51